torchinductor/codegen/autotuner.py

```diff
@@ -0,0 +1,166 @@
+#     Decorator for auto-tuning a function.
+
+#     .. highlight:: python
+#     .. code-block:: python
+
+#         @triton.autotune()
+#         wrap_conv(x, w, bias, stride, ...)
+
+#     :note: For the wrap function, the candidate kernels will run multiple time.
+#     """
+#     def inner(fn):
+#         autotuner = Autotuner(fn)
+#         return autotuner
+
+#     return inner
+
+
+class Autotuner:
+    def __init__(self):
+
+        self.cache = dict()
+
+    def _bench(self, *args, kernel, **kwargs):
+        def kernel_call():
+            kernel(*args, **kwargs)
+
+        return triton.testing.do_bench(kernel_call, warmup=10, rep=50)
+
+    # def __call__(self, *args, **kwargs):
+    #     # collect input shape/stride/device, layer params
+    #     self.fn.set_args(*args, **kwargs)
+    #     # get candidate kernels for fn
+    #     self.kernels = self.fn.candidate_kernels()
+    #     # filter kernels that args/kwargs does not meet requirements
+    #     self.fn.filter_kernels(self.kernels)
+    #     # if only one choice, return that kernel
+    #     if len(self.kernels) == 1:
+    #         kernel = self.kernels[0]
+    #         return kernel(*args, **kwargs)
+    #     # use input shape, stride, layer parameters as key to cache the best kernel to use
+    #     key = self.fn.gen_key()
+    #     if key not in self.cache:
+    #         bench_start = time.time()
+    #         timings = {
+    #             kernel: self._bench(*args, kernel=kernel, **kwargs) for kernel in self.kernels
+    #         }
+    #         bench_end = time.time()
+    #         self.bench_time = bench_end - bench_start
+    #         self.cache[key] = builtins.min(timings, key=timings.get)
+    #         self.kernels_timings = timings
+    #         kernel = self.cache[key]
+    #     self.best_kernel = kernel
+    #     return self.best_kernel(*args, **kwargs)
+
+
+autotune = Autotuner()
+
+
+def tuned_conv(*args, **kwargs):
+
+    # gather info
+    x = args[0]
+    w = args[1]
+    # bias = args[2]
+    stride = args[3]
+    padding = args[4]
+    dilation = args[5]
+    transposed = args[6]
+    output_padding = args[7]
+    groups = args[8]
+
+    BATCH, IN_C, IN_H, IN_W = x.shape
+    KERNEL_N, _, KERNEL_H, KERNEL_W = w.shape
+    stride = stride
+    padding = padding
+    dilation = dilation
+    transposed = transposed
+    output_padding = output_padding
+    groups = groups
+    stride_x = x.stride()
+    stride_w = w.stride()
+    # the identifiable args for the layers
+    id_args = [
+        BATCH,
+        IN_C,
+        IN_H,
+        IN_W,
+        KERNEL_N,
+        KERNEL_H,
+        KERNEL_W,
+        stride,
+        padding,
+        dilation,
+        transposed,
+        output_padding,
+        groups,
+        stride_x,
+        stride_w,
+    ]
+    use_cuda = x.is_cuda
+
+    # gen_key
+    key = tuple([arg for arg in id_args])
+    key = ("conv",) + key
+
+    # candidate kernels
+    kernels = ["aten.convolution"]
```

Comment:
I think we can just directly puts the kernels here and don't need strings + kernel_dict

Reply:
If we want to print the selected kernels for debugging, aten.convolution.__name__( = 'convolution') or triton_ops.conv.__name__ or triton_ops.conv1x1.__name__ is not meaningful enough.